### PR TITLE
[RHBPMS-4312] Key store password reading made more general to support

### PR DESCRIPTION
**Pull request by rstancel**
Jira: https://issues.jboss.org/browse/RHBPMS-4312

This is enhancing previous PR: https://github.com/kiegroup/droolsjbpm-integration/pull/1273 with possibility to read multiple passwords

### CODE REVIEW

**Comment by rstancel** [comment]
@mswiderski can you please review this? Please also notice that previous PR is not merged yet

**Comment by rstancel** [comment]
this PR is cooperating with following PRs: https://github.com/kiegroup/drools/pull/1619 and https://github.com/kiegroup/kie-docs/pull/509

File: kie-server-parent/kie-server-common/src/main/java/org/kie/server/common/KeyStoreHelperUtil.java [hunk 1]
```diff
@@ -7,14 +7,44 @@
...
+
+    // the private key identifier for controller
+     private static final String PROP_PWD_CTRL_ALIAS = "kie.keystore.key.ctrl.alias";
+    // the private key identifier for controller
+    private static final String PROP_PWD_CTRL_PWD = "kie.keystore.key.ctrl.pwd";
+
     private static final Logger logger = LoggerFactory.getLogger(KeyStoreHelperUtil.class);
 
-    public static String loadPassword(KieServerConfig config) {
+    public static String loadRestControllerPassword() {
```
Comment (mswiderski) [inline kie-server-parent/kie-server-common/src/main/java/org/kie/server/common/KeyStoreHelperUtil.java]: this is not really controller password but kie server 

in general kie server can call controller and here is the password for controller required - meaning one that is configured in controller's security domain). Then controller can also call kie servers and here the password for kie server is required - meaning one that is configured in kie server's security domain.

So I'd rename the method to loadServerPassword and loadControllerPassword

File: kie-server-parent/kie-server-common/src/main/java/org/kie/server/common/KeyStoreHelperUtil.java [hunk 2]
```diff
@@ -7,14 +7,44 @@
 import org.slf4j.LoggerFactory;
 
 public class KeyStoreHelperUtil {
+    // the private key identifier for REST controller
+    private static final String PROP_PWD__REST_ALIAS = "kie.keystore.key.rest.alias";
+    // the private key identifier for REST controller
+    private static final String PROP_PWD__REST_PWD = "kie.keystore.key.rest.pwd";
```
Comment (mswiderski) [inline kie-server-parent/kie-server-common/src/main/java/org/kie/server/common/KeyStoreHelperUtil.java]: same applies here for properties - there is either server (kie server) password or controller password.

**Comment by mswiderski** [comment]
@rstancel maybe it would make sense to combine this PR with #1273 as both require small changes

**Comment by rstancel** [comment]
@mswiderski I have updated the code based on your comments

**Comment by mswiderski** [comment]
@rstancel just merged #1273 and seems it conflicts with this one, could you please resolve and then it will be ready for merge, thanks

**Comment by rstancel** [comment]
@mswiderski Maciej, conflicts has been resolved

**Pull request merged**